app/models/spree/product_decorator.rb

```diff
@@ -0,0 +1,35 @@
+module Spree
+  module ProductDecorator
+    def self.prepended(base)
+      base.after_create :create_shopify_product
+      base.after_update :update_shopify_product
+      base.after_destroy :destroy_shopify_product
+    end
+
+    private
+
+    def create_shopify_product
+      service = Shopify::ProductExporter.new(spree_product: self)
+      service.perform
+    end
+
+    def update_shopify_product
+      if assembly?
```

Comment:
I am unsure if this should be done in some kind of factories, what do you guys think?